documentation/src/main/asciidoc/performance_guide/performance_guide.asciidoc

```diff
@@ -0,0 +1,232 @@
+= Infinispan {infinispanversion} Performance Guide
+The Infinispan community
+:toc2:
+:icons: font
+:toclevels: 3
+:numbered:
+:stem:
+
+This guide will give you information and tweaks about tuning Infinispan performance (both server and library mode).
+
+== Capacity planning
+
+Infinispan stores data in memory. Depending on operating mode data is stored in serialized form (server mode) or directly
+directly in memory (library mode). Data size can be estimated using sophisticated tools like  http://openjdk.java.net/projects/code-tools/jol/[Java Object Layout]
+and total amount of required memory can be calculated using the following formulas:
+
+[stem]
+.Total Data Set in library mode
+++++
+Total Data Set = Number Of Entries * (Key Size + Value Size + 200 b (Me tadata))
+++++
+
+[stem]
+.Total Data Set in server mode
+++++
+Total Data Set = Number Of Entries * (Serialized Key Size + Serialized Value Size + 200 b (Me tadata))
+++++
+
+
+In case of Local or Replicated mode, all data needs to fit in memory, so calculating the amount of required memory is trivial.
+
+Calculating memory requirements for Distributed mode is slightly more complicated and requires using the following:
+
+[stem]
+.Required memory in Distributed mode
+++++
+Required Memo ry = (Total Data Set)/(Nodes)*(1+(Node Failures)/(Nodes - Node Failures))
```

Comment:
I'm personally OK with this. However since we are suggesting something else than for the produce I would like to hear @alanfx and @tristantarrant opinion.